src/query/scan_manager.c

```diff
@@ -2379,11 +2379,26 @@ scan_get_index_oidset (THREAD_ENTRY * thread_p, SCAN_ID * s_id, DB_BIGINT * key_
 	  assert (er_errid () != NO_ERROR);
 	  goto exit_on_error;
 	}
-      ret = btree_range_scan (thread_p, bts, btree_range_scan_select_visible_oids);
-      if (ret != NO_ERROR)
+      bool retry = true;
```

Comment:
@adrianbrab 
I do not think this handling is needed.
I have added generic handling in function `btree_range_scan` for all cases where `btree_range_scan_select_visible_oids` is used as key handling/processing function (`btree_range_scan_select_visible_oids` is only used in this context as argument for `btree_range_scan`).